homeassistant/components/efergy/sensor.py

```diff
@@ -32,11 +42,26 @@
 DEFAULT_UTC_OFFSET = "0"
 
 SENSOR_TYPES = {
-    CONF_INSTANT: ["Energy Usage", POWER_WATT],
-    CONF_AMOUNT: ["Energy Consumed", ENERGY_KILO_WATT_HOUR],
-    CONF_BUDGET: ["Energy Budget", None],
-    CONF_COST: ["Energy Cost", None],
-    CONF_CURRENT_VALUES: ["Per-Device Usage", POWER_WATT],
+    CONF_INSTANT: {
+        ATTR_NAME: "Energy Usage",
+        ATTR_UNIT_OF_MEASUREMENT: POWER_WATT,
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        ATTR_STATE_CLASS: STATE_CLASS_TOTAL_INCREASING,
```

Comment:
This might be wrong.

Reply:
It is, as power usage in Watts is actually a current value. If this sensor is really providing Watts, the name of this sensor is wrong, as Energy is not expressed in Watts, Watts represents power.

Reply:
We discussed the name change previously but decided to leave that for another change as it might turn into a breaking change.
https://github.com/home-assistant/core/issues/54005#issuecomment-894240104
I am happy to fix the naming and make them more correct and useful. If we are going to fix the names, is there a standard we should follow for "electricity" sensors?

Regarding the state class STATE_CLASS_TOTAL_INCREASING I don't think is correct for efergy as the amount sensor is by period, so at some point DAY/MONTH/YEAR it will reset and start from 0.

Reply:
> If we are going to fix the names, is there a standard we should follow for "electricity" sensors?

The dictionary? (Sorry it sounds blunt maybe, I don't mean to, these terms are are well established and documented in general).

> Regarding the state class STATE_CLASS_TOTAL_INCREASING I don't think is correct for efergy as the amount sensor is by period, so at some point DAY/MONTH/YEAR it will reset and start from 0.

As said, this is a sensor representing POWER, you can accumulate power over time, thus `STATE_CLASS_TOTAL_INCREASING` and `STATE_CLASS_TOTAL` are both not viable. This change is incorrect.

Reply:
> @Data-Monkey Try the latest commit and see what you can do if its not working.

Hi, @tkdrob, I just tried the new commit, but I guess the new STATE_CLASS_* are not available yet in 2021.8.7. (or I am doing something wrong)
I get this error: 
`ImportError: cannot import name 'STATE_CLASS_TOTAL_INCREASING' from 'homeassistant.components.sensor' (/usr/src/homeassistant/homeassistant/components/sensor/__init__.py)`

Reply:
@Data-Monkey this is dev, you cannot test this change against a current stable release, only against the `dev` branch.

Reply:
> The dictionary? (Sorry it sounds blunt maybe, I don't mean to, these terms are are well established and documented in general).

Hi @frenck, first of all, thanks for all the effort you put into this!

yes, I agree the terms are well established and all your remarks are understood. What I was asking is, is there a naming standard that all energy integrations should adhere to? 

Power Usage vs Power Consumption vs Current Power vs ...
Total Energy Consumption vs Energy Consumption vs Energy Used vs Energy Demand  ...
Should the platform name be referenced in the sensor name? sensor.efergy_ 

looking at the code of some other integrations I don't see a naming standard
if there is no standard then I will just suggest some names

Reply:
> yes, I agree the terms are well established and all your remarks are understood. What I was asking is, is there a naming standard that all energy integrations should adhere to?

No, there is not, I was only pointing out it refers to energy at a point, while it is power. Power and energy are not the same things. Its like naming a living room temperature sensor: "Radiation from the sun". It simply doesn't match.